src/request-sender.spec.ts

```diff
@@ -115,7 +117,18 @@ describe('RequestSender', () => {
         });
 
         it('creates a HTTP request with CSRF token if it exists', () => {
-            jest.spyOn(cookie, 'get').mockImplementation(key => key === 'XSRF-TOKEN' ? 'abc' : undefined);
+            const mockFn = (key: string) => key === 'XSRF-TOKEN' ? 'abc' : undefined;
+
+            /*
+             * jest.Mocked has a bug with overloads
+             *
+             * The change that broke this was in @types/jest@23.3.14:
+             * https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34889
+             *
+             * The spyOn type is using ReturnType and when used with overloads
+             * it chooses the last overload.
+             */
+            jest.spyOn(cookie, 'get').mockImplementationOnce(mockFn as typeof cookie.get);
```

Comment:
I'm not if the comment above refers to the fact that you have to cast it as `typeof cookie.get` or you're switching to `mockImplementationOnce`? If possible, I'd prefer to keep `mockImplementation` because that's the intended scenario we are trying to mock.

Reply:
The comment is in regards to the `typeof cookie.get`. I'll update the comment to better reflect this.

I will also update to `mockImplementation`. `mockImplementationOnce` was a snowflake from a previous implementation of mocking the module, in which I tried.